src/components/notebook-utils.js

```diff
@@ -246,10 +249,20 @@ export const AnalysisDuplicator = ({ destroyOld = false, fromLauncher = false, p
       onClick: async () => {
         setProcessing(true)
         try {
-          await (destroyOld ?
-            Ajax().Buckets.analysis(googleProject, bucketName, printName, toolLabel).rename(newName) :
-            Ajax().Buckets.analysis(googleProject, bucketName, getFileName(printName), toolLabel).copy(`${newName}.${getExtension(printName)}`, bucketName, !destroyOld)
-          )
+          const rename = !!googleProject ?
+            () => Ajax().Buckets.analysis(googleProject, bucketName, printName, toolLabel).rename(newName) :
+            () => Ajax().AzureStorage.blob(workspaceId, printName).rename(newName)
+
+          const duplicate = !!googleProject ?
+            () => Ajax().Buckets.analysis(googleProject, bucketName, getFileName(printName), toolLabel).copy(`${newName}.${getExtension(printName)}`, bucketName, !destroyOld) :
```

Comment:
Here, destroyOld is being used to determine if we should clear the metadata of the bucket object. To be fair though, its a bit confusing since it we are duplicating the cloud storage object we always want to clear the metadata. I refactored it a bit.